packages/block-library/src/cover/edit.js

```diff
@@ -311,11 +314,43 @@ function CoverEdit( {
 		minHeight,
 		minHeightUnit,
 		style: styleAttribute,
-		url,
 		alt,
 		allowedBlocks,
 		templateLock,
 	} = attributes;
+
+	let { url, backgroundType } = attributes;
+
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const media = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+	const mediaUrl = media?.source_url;
+
+	useEffect( () => {
+		if ( useFeaturedImage ) {
+			setAttributes( {
+				dimRatio: dimRatio === 100 ? 50 : dimRatio,
```

Comment:
What's the logic here?

Reply:
If the user starts with a solid color and then toggles the featured image the color becomes an overlay dimmed by 50%, it is the current behaviour but for the featured image toggle, as we don't have a media object.

Reply:
Why this behavior is different between featured image and regular image.

Reply:
Because the regular image is a media object handled by `onSelectMedia`, while featured image is just pretending to have an image in `Edit`  replacing `url` with `mediaUrl`.

Reply:
Ok so in fact we already have the same behavior, it's just done in a different place. That said, I'd probably avoid doing this in an effect like done here because it will create two undo levels, instead I'd move this logic to the "toggleFeaturedImage" callback and only have one `setAttributes` call.